olp-cpp-sdk-authentication/src/IntrospectAppResultImpl.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "IntrospectAppResultImpl.h"
+
+#include "Constants.h"
+#include "olp/core/http/HttpStatusCode.h"
+
+namespace olp {
+namespace authentication {
+
+IntrospectAppResultImpl::IntrospectAppResultImpl() noexcept
+    : IntrospectAppResultImpl(http::HttpStatusCode::SERVICE_UNAVAILABLE,
+                              Constants::ERROR_HTTP_SERVICE_UNAVAILABLE) {}
+
+IntrospectAppResultImpl::IntrospectAppResultImpl(
+    int status, std::string error,
+    std::shared_ptr<rapidjson::Document> json_document) noexcept
+    : BaseResult(status, error, json_document) {
+  // Extra response data if no errors reported
+  if (!HasError()) {
+    if (!IsValid()) {
+      status_ = http::HttpStatusCode::SERVICE_UNAVAILABLE;
+      error_.message = Constants::ERROR_HTTP_SERVICE_UNAVAILABLE;
+    } else {
+      if (json_document->HasMember(Constants::CLIENT_ID))
+        client_id_ = (*json_document)[Constants::CLIENT_ID].GetString();
+      if (json_document->HasMember(Constants::NAME))
+        name_ = (*json_document)[Constants::NAME].GetString();
+      if (json_document->HasMember(Constants::DESCRIPTION))
+        description_ = (*json_document)[Constants::DESCRIPTION].GetString();
+      if (json_document->HasMember(Constants::CREATED_TIME))
+        created_time_ = (*json_document)[Constants::CREATED_TIME].GetInt64();
+      if (json_document->HasMember(Constants::REALM))
+        realm_ = (*json_document)[Constants::REALM].GetString();
+      if (json_document->HasMember(Constants::TIER))
+        tier_ = (*json_document)[Constants::TIER].GetString();
+      if (json_document->HasMember(Constants::HRN))
+        hrn_ = (*json_document)[Constants::HRN].GetString();
```

Comment:
Prefer curly braces. Even though it's allowed to omit curly braces for simple conditionals, these conditionals are not trivial and hard to read without curly braces